boards/grand_central_m4/src/pins.rs

```diff
@@ -0,0 +1,749 @@
+//! Grand Cental M4 Express pins
+
+use super::{hal, pac::MCLK, 
+    pac::SERCOM0, pac::SERCOM1, //pac::SERCOM2, 
+    pac::SERCOM3, pac::SERCOM4, pac::SERCOM5, 
+    pac::SERCOM6, pac::SERCOM7, target_device};
+
+use hal::define_pins;
+use hal::gpio::{self, *};
+use hal::sercom::{
+    PadPin,
+    I2CMaster3, Sercom3Pad0, Sercom3Pad1,
+    I2CMaster6, Sercom6Pad0, Sercom6Pad1,
+    SPIMaster7, //Sercom7Pad3,Sercom7Pad0, Sercom7Pad1,
+    UART0, Sercom0Pad1, Sercom0Pad0,
+    UART4, Sercom4Pad0, Sercom4Pad1,
+    UART1, Sercom1Pad0, Sercom1Pad1,
+    UART5, Sercom5Pad0, Sercom5Pad1,
+};
+use hal::time::Hertz;
+
+use hal::clock::GenericClockController;
+
+#[cfg(feature = "usb")]
+use hal::usb::usb_device::bus::UsbBusAllocator;
+#[cfg(feature = "usb")]
+pub use hal::usb::UsbBus;
+
+define_pins!(
+    /// Maps the pins to their arduino names and
+    /// the numbers printed on the board.
+    struct Pins,
+    target_device: target_device,
+
+    /// LED Pins
+    pin tx_led = c30,
+    pin rx_led = c31,
+    //pin led = b1,
+    pin neopixel = c24,
+
+    /// Pin A0 (analog). Can act as a true analog output as it has a DAC (which is
+    /// currently not supported by this hal) as well as input.
+    pin a0 = a2,
+    /// Pin A1 (analog). Can act as a true analog output as it has a DAC (which is
+    /// currently not supported by this hal) as well as input.
+    pin a1 = a5,
+    /// Pin A2 (analog)
+    pin a2 = b3,
+    /// Pin A3 (analog)
+    pin a3 = c0,
+    /// Pin A4 (analog), PWM capable
+    pin a4 = c1,
+    /// Pin A5 (analog), PWM capable
+    pin a5 = c2,
+    pin a6 = c3,
+    pin a7 = b4,
+    pin a8 = b5,
+    pin a9 = b6,
+    pin a10 = b7,
+    pin a11 = b8,
+    pin a12 = b9,
+    pin a13 = a4,
+    pin a14 = a6,
+    pin a15 = a7,
+
+    /// Pin 0 (digital), aka RX for Serial1. PWM capable.
+    //pin d0 = b25,
+    /// Pin 1 (digital), aka TX for Serial1. PWM capable.
+    //pin d1 = b24,
+    /// Pin 2, PWN capable, analog input capable
+    pin d2 = c18,
+    /// Pin 3
+    pin d3 = c19,
+    /// Pin 4, PWM capable
+    pin d4 = c20,
+    /// Pin 5. Output-only with rail-to-rail HI level. PWM capable.
+    pin d5 = c21,
+    pin d6 = d20,
+    /// Pin 7, PWM capable
+    pin d7 = d21,
+    pin d8 = b18,
+    /// Pin 9, PWM capable
+    pin d9 = b2,
+    /// Pin 10, PWM capable
+    pin d10 = b22,
+    /// Pin 11, PWM capable
+    pin d11 = b23,
+    /// Pin 12, PWM capable
+    pin d12 = b0,
+    /// Pin 13, PWM capable. Also attached to onboard LED.
+    pin d13 = b1,
+    //pin d14 = b16,
```

Comment:
Lots of stuff commented out in this file, could you explain why?

Reply:
Sure, the commented pins are those that are defined with friendlier names elsewhere. Looking at the other boards it seemed that most preferred the friendlier name (i.e. uart0_rx instead of d0). I'm adding comments to make this clear. I'll push a new commit soon.

Reply:
I cleaned up the pin documentation and did some minor refactoring. I also noticed that putting Digital in the pin Set was not consistent between boards. I choose to remove it.